### PR TITLE
V2v structure

### PR DESCRIPTION
**Pull request by maansthoernvik**

### CODE REVIEW

**Comment by Rashadse** [comment]
now*

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
-
+*.tar
```
Comment (Rashadse) [inline .gitignore]: why won't we have tar files on our repo?

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+include_directories(SYSTEM ${CMAKE_BINARY_DIR})
+
+# Executables
+add_executable(${PROJECT_NAME} ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp ${CMAKE_BINARY_DIR}/messages.cpp ${V2V_DIR}/v2v.cpp)
+target_link_libraries(${PROJECT_NAME} ${CLUON_LIBRARIES})
+
+# Testing
+#enable_testing()
+#add_executable(${PROJECT_NAME}-test ${TEST_DIR}/UnitTests.cpp <File to test>)
+#add_test(NAME ${PROJECT_NAME}-test COMMAND ${PROJECT_NAME}-test)
```
Comment (Rashadse) [inline CMakeLists.txt]: are these lines intended to be left out and commented like this?

File: v2v/v2v.cpp [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                                                            AnnouncePresence ap = cluon::extractMessage<AnnouncePresence>(std::move(envelope));
+                                                            std::cout << "received 'AnnouncePresence' from '"
+                                                                      << ap.vehicleIp() << ":" << ap.activePort() << "', GroupID '"
+                                                                      << ap.groupId() << "'!" << std::endl;
+
+                                                            /* TODO: monitor these IPs so you know which ones to choose from */
+
+                                                            break;
+                                                        }
+                                                        default: std::cout << "¯\\_(ツ)_/¯" << std::endl;
```
Comment (Rashadse) [inline v2v/v2v.cpp]: lel

File: v2v/v2v.cpp [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+                                                             std::cout << "received '" << followRequest.LongName()
+                                                                       << "' from '" << sender << "'!" << std::endl;
+
+                                                             // After receiving a FollowRequest, check first if there is currently no car already following.
+                                                             if (followerIp.empty()) {
+                                                                 followerIp = sender; // If no, add the requester to known follower slot and establish a
+                                                                 // sending channel.
+                                                                 toFollower = std::make_shared<cluon::UDPSender>(followerIp, DEFAULT_PORT);
+                                                                 followResponse();
+                                                             }
```
Comment (Rashadse) [inline v2v/v2v.cpp]: the indentation in this file makes it really hard to read the code on GitHub

File: v2v/v2v.cpp [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+ */
+void V2VService::followResponse() {
+    if (followerIp.empty()) return;
+    FollowResponse followResponse;
+    followResponse.ntpServerIp(NTP_IP);
+    toFollower->send(encode(followResponse));
+}
+
+/**
+ * This function sends a StopFollow (id = 1004) request on the ip address of the parameter vehicleIp. If the IP address is neither
```
Comment (Rashadse) [inline v2v/v2v.cpp]: this line is too long

File: build_local.sh [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+# Author: Rashad Kamsheh
+# This script builds an image for the current platform from which it is being build. Building using this script on the
+# VM and then transfering the built image to the car will not work.
+
```
Comment (Rashadse) [inline build_local.sh]: maybe add a way to stop images and removed them
i suggest 
docker stop $(docker ps -a -q)
docker rm $(docker ps -a -q)
docker rmi group7/carcode

File: main.cpp [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#include <iostream>
+
+#include "v2v.hpp"
+
+
+int main(int /*argc*/, char** /*argv*/) {
+    std::cout << "Ran" << std::endl;
```
Comment (Rashadse) [inline main.cpp]: what is the purpose of this main file

File: .gitignore (hunk 1 above)
Comment (maansthoernvik) [reply .gitignore]: They're large and not part of source code

File: CMakeLists.txt (hunk 2 above)
Comment (maansthoernvik) [reply CMakeLists.txt]: Until we have tests, yes

File: v2v/v2v.cpp (hunk 4 above)
Comment (maansthoernvik) [reply v2v/v2v.cpp]: Fixed

File: v2v/v2v.cpp (hunk 5 above)
Comment (maansthoernvik) [reply v2v/v2v.cpp]: Fixed

File: build_local.sh (hunk 6 above)
Comment (maansthoernvik) [reply build_local.sh]: Sounds good, but we won't want to remove them every time

File: main.cpp (hunk 7 above)
Comment (maansthoernvik) [reply main.cpp]: Just to see that the docker image "ran". This is where we would instantiate our V2VService etc.

File: .gitignore (hunk 1 above)
Comment (madiseniman) [reply .gitignore]: But don't we need them?

Comment (maansthoernvik) [reply .gitignore]: We don't add them for the same reason we don't add the build folder and compiled code, they change too often.

**Pull request merged**